### ISSUE TITLE
semantic highlighting doesn't take effect immediately after pasting code

### ISSUE THREAD

**Issue by Eskibear** [redhat-developer/vscode-java#1421]
When you cut and paste code, semantic highlighting may sometimes take no effect.
 
I did take a quick look, when I inspect token types, it's correct. **So I suspect this might be a vscode issue.**  Below is a screenshot (the theme selectors are too long, so I manually removed them). 
![image](https://user-images.githubusercontent.com/2351748/80679217-07bf0f00-8aef-11ea-8e2d-ef6733330091.png)

#### Steps to reproduce:

* Open an empty folder and create a new file `foo/bar/apiImpl/Hello.java`:
```java
package foo.bar.apiImpl;
public class Hello { }
```

* Semantic highlighting should be working well, you can observer `apiImpl`.
* Cut the first line (package.....), type some whitespaces, and paste it back.
* Boom! Now you see the color fallbacks to syntax highlighting.  
* If you modify the content, e.g. appending a whitespace to any line, the color would get back to work. 

vscode-java: 0.61.0
VSCode version:
```
Version: 1.44.2
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T17:07:18.473Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.4.0
```

**Comment by Eskibear** [comment]
I can also reproduce it in latest vscode-insiders
```
Version: 1.45.0-insider
Commit: a0fe37870c42e0704a492cdc6b8550b4cf23f63c
Date: 2020-04-30T05:36:26.465Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.4.0
```

**Comment by Eskibear** [comment]
fixed by #1435
